File: android/src/main/kotlin/com/example/bheya_network/models/common/Network.java
package com.example.bheya_network.models.common;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class Network implements Serializable {

    @SerializedName("mcc")
    private String mcc;
    @SerializedName("mnc")
    private String mnc;
    @SerializedName("iso")
    private String iso;

    public Network() {
    }

    public String getMcc() {
        return mcc;
    }

    public void setMcc(String mcc) {
        this.mcc = mcc;
    }

    public String getMnc() {
        return mnc;
    }

    public void setMnc(String mnc) {
        this.mnc = mnc;
    }

    public String getIso() {
        return iso;
    }

    public void setIso(String iso) {
        this.iso = iso;
    }
}

File: lib/models/common/network.dart
class Network {
  String? iso;
  String? mcc;
  String? mnc;

  Network({this.iso, this.mcc, this.mnc});

  Network.fromJson(Map<String, dynamic> json) {
    iso = json['iso'];
    mcc = json['mcc'];
    mnc = json['mnc'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['iso'] = iso;
    data['mcc'] = mcc;
    data['mnc'] = mnc;
    return data;
  }
}
